components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -111,6 +112,22 @@ pub fn locate_binary_2_args(substr: &Option<Bytes>, s: &Option<Bytes>) -> Result
         .or(Some(0)))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn un_hex(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(arg
```

Comment:
Looks like you can implement `Vec::from_hex` manually in order to avoid this clone.

Reply:
> Looks like you can implement `Vec::from_hex` manually in order to avoid this clone.

Because to change 'bytes', need to take ownership, is that ok? can u give me an example?

Reply:
That's why you may implement the algorithm (unhex) manually in order to not to take the ownership. You can directly access the **byte slice** instead of the byte ownership. e.g. 

```rust
if slice.len() % 2 == 1 {
    // handle the first 1 byte as a hex.
    // handle the rest bytes 2 by 2 as hex.
} else {
    // handle bytes 2 by 2 as hex.
}
```

Reply:
Thank you so much! I am a rust beginner, if anything is wrong,please comment